include/nlohmann/json.hpp

```diff
@@ -1977,11 +2168,11 @@ class basic_json
 
         if (indent >= 0)
         {
-            s.dump(*this, true, ensure_ascii, static_cast<unsigned int>(indent));
+            s.dump(*this, true, ensure_ascii, static_cast<unsigned int>(indent), serialize_binary);
```

Comment:
The `serialize_binary` parameter should be part of the signature of `dump`. I think a `bool` may be insufficient, because we may want to have similar behavior as `error_handler` controls for invalid UTF-8:

- strict - throw on binary values
- ignore - ignore binary values (i.e., do not serialize them)
- replace - dump with some kind of serialization

I'm not sure whether the exact same `error_handler_t` enum should be used, but something like this could do the job.